magento/magento2 #2370: A PHPDoc comment typo fix

cavalier79 opened the pull request:
Fixed typo in 
app/code/Magento/Config/Model/Config/SourceFactory.php on line 26

wrong typo was 

@return mixed 

fixed in 

@return \Magento\Framework\Option\ArrayInterface|object

Commit:
magento/magento2 #2370: A PHPDoc comment for \Magento\Config\Model\Config\SourceFactory::create() method is wrong

Fixed type in return comment

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/signed)](https://cla.dev.magento.com/magento/magento2?pullRequest=8758) <br/>All committers have signed the CLA.

Commit:
magento/magento2 #2370:

A PHPDoc comment for \Magento\Config\Model\Config\SourceFactory::create() method is wrong
The correct return parameter is  \Magento\Framework\Option\ArrayInterface

ishakhsuvarov commented:
@cavalier79 Initial commits were merged to the `develop` branch. Thank you for contribution!
